### ISSUE TITLE
IExecutionContextManager method isn't very descriptive

### ISSUE THREAD

**Issue by smithmx** [dolittle-fundamentals/DotNET.Fundamentals#201]
When we need to set the context (usually in service classes which aren't per request lifetime) then we call:

_executionContextManager.CurrentFor(tenantId, some_correlation_id);

It feels out of place and it isn't immediately obvious when it's being called (or that setting is a side-effect of getting the "current").

How about something like 

using(_executionContextManager.SetScope(tenantId,correlationId))
{
    //do stuff
}

It's not a disposable in the strict sense but it helps reinforce what's happening.

**Comment by einari** [comment]
Testing comments from Asana

**Comment by einari** [comment]
Does the comment go back?

**Comment by einari** [comment]
Agree. Name has much to be desired. Having it scoped with a `using` statement is not something that would be viable for some of the places we need it. It basically sets the current async context and relies on anything in that context to now be relative to the scope configured. But in some cases it could be very useful to have the possibility to enter a scope and exit it, an object implementing `IDisposable` would then keep track of the scope before and could pop it back when being disposed of.
